lib/HooksHandler.php

```diff
@@ -68,16 +68,16 @@ class HooksHandler {
 	private $userNotificationConfigHandler;
 
 	public function __construct(
-		IConfig $config = null,
-		Engine $engine = null,
-		IHasher $hasher = null,
-		ITimeFactory $timeFactory = null,
-		IL10N $l10n = null,
-		PasswordExpired $passwordExpiredRule = null,
-		OldPasswordMapper $oldPasswordMapper = null,
-		ISession $session = null,
-		IManager $notificationManager = null,
-		UserNotificationConfigHandler $userNotificationConfigHandler = null
+		?IConfig $config,
```

Comment:
```suggestion
		?IConfig $config = null,
```

we can leave the default `= null` so that callers are still allowed to call the constructor without specifying any parameters.

(same for other lines changed)